src/classifier/Features.h

```diff
@@ -50,4 +55,76 @@ class Features {
   Pipe *pipe_; // The pipe that owns this feature handler.
 };
 
+
+//struct  FeatureLabelPair : Structure to define a a feature-label pair
+struct FeatureLabelPair {
+	uint64_t feature;
+	int label;
+};
+
+//struct  FeatureLabelPairMapper: Structure to define a hash function and a comparison function for FeatureLabelPair
+struct FeatureLabelPairMapper {
+	template <typename TSeed>
+	inline void HashCombine(TSeed value, TSeed *seed) const {
+		*seed ^= value + 0x9e3779b9 + (*seed << 6) + (*seed >> 2);
+	}
+	//Hash function
+	inline size_t operator()(const FeatureLabelPair& p) const {
+		size_t hash = std::hash<uint64_t>()(p.feature);
+		size_t hash_2 = std::hash<int>()(p.label);
+
+		HashCombine<size_t>(hash_2, &hash);
+		return hash;
+	}
+	//Comparison function
+	inline bool operator()(const FeatureLabelPair &p, const FeatureLabelPair &q) const {
+		return p.feature == q.feature && p.label == q.label;
+	}
+};
+
+//Defines a hash-table of FeatureLabelPair keys with values, of double type
+typedef std::unordered_map<FeatureLabelPair, double, FeatureLabelPairMapper, FeatureLabelPairMapper > FeatureLabelPairHashMap;
+
+//class  FeatureLabelCache: Hash-table for caching FeatureLabelPair keys with corresponding values
+class FeatureLabelCache{
+
+public:
+	FeatureLabelCache() {
+		hits_ = 0;
+		misses_ = 0;
+	};
+	virtual ~FeatureLabelCache() {};
+	
+	int hits() const { return hits_; };
+	int misses() const { return misses_; };
+	int size() const { return cache_.size(); };
```

Comment:
As a rule of thumb, function names should be capitalized, unless they are simply get methods for variable members. So hits() and misses() and fine, but size() is not.